osutil/kcmdline/kcmdline.go

```diff
@@ -441,3 +445,39 @@ func (kap *ArgumentPattern) UnmarshalYAML(unmarshal func(interface{}) error) err
 
 	return nil
 }
+
+func (kap *ArgumentPattern) marshalToString() (string, error) {
+	return fmt.Sprintf(`%s=%s`, kap.param, kap.value.String()), nil
+}
+
+func (kap *ArgumentPattern) UnmarshalBinary(data []byte) error {
+	return kap.unmarshalFromString(string(data))
+}
+
+func (kap *ArgumentPattern) MarshalBinary() ([]byte, error) {
+	str, err := kap.marshalToString()
+	if err != nil {
+		return []byte{}, err
+	}
+	return []byte(str), nil
+}
+
+func (kap *ArgumentPattern) UnmarshalText(data []byte) error {
+	return kap.unmarshalFromString(string(data))
+}
+
+func (kap *ArgumentPattern) MarshalText() ([]byte, error) {
+	str, err := kap.marshalToString()
+	if err != nil {
+		return []byte{}, err
+	}
+	return []byte(str), nil
+}
+
+func (kap *ArgumentPattern) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var arg string
+	if err := unmarshal(&arg); err != nil {
+		return errors.New("cannot unmarshal kernel argument")
```

Comment:
Should this include the underlying error?
```suggestion
		return fmt.Errorf("cannot unmarshal kernel argument: %v", err)
```

Reply:
Good point.

Reply:
I had forgot to fix this one. Now it is pushed.